1.16.5/src/main/java/io/github/axolotlclient/mixin/PlayerListHudMixin.java

```diff
@@ -150,4 +165,156 @@ public abstract class PlayerListHudMixin {
+		boolean bl, int n, int o, int p, int q, int r, List list2, int t, int u, int s, int v, int y, int z, PlayerListEntry playerListEntry2
+	) {
+		if (!BedwarsMod.getInstance().isEnabled() || !BedwarsMod.getInstance().isWaiting()) {
+			return;
+		}
+		int startX = v + i + 1;
+		int endX = startX + n;
+		String render;
+		try {
+			if (playerListEntry2.getProfile().getName().contains(Formatting.OBFUSCATED.toString())) {
+				return;
+			}
+
+			render = String.valueOf(HypixelAbstractionLayer.getPlayerLevel(playerListEntry2
+					.getProfile().getId().toString().replace("-", ""),
+				LevelHeadMode.BEDWARS.toString()));
+		} catch (Exception e) {
+			return;
+		}
+		this.client.textRenderer.drawWithShadow(matrixStack,
+			render,
+			(float) (endX - this.client.textRenderer.getWidth(render)) + 20,
+			(float) y,
+			-1
+		);
+	}
+
+	@Inject(
+		method = "renderScoreboardObjective",
+		at = @At(
+			value = "INVOKE",
+			target = "Lnet/minecraft/client/font/TextRenderer;drawWithShadow(Lnet/minecraft/client/util/math/MatrixStack;Ljava/lang/String;FFI)I",
+			ordinal = 1
+		),
+		cancellable = true
+	)
+	private void axolotlclient$renderCustomScoreboardObjective(
+		ScoreboardObjective objective, int y, String player, int startX, int endX,
+		PlayerListEntry playerEntry, MatrixStack matrices, CallbackInfo ci
+	) {
+		if (!BedwarsMod.getInstance().isEnabled()) {
+			return;
+		}
+
+		BedwarsGame game = BedwarsMod.getInstance().getGame().orElse(null);
+		if (game == null) {
+			return;
+		}
+		BedwarsPlayer bedwarsPlayer = game.getPlayer(playerEntry.getProfile().getName()).orElse(null);
+		if (bedwarsPlayer == null) {
+			return;
+		}
+		ci.cancel();
+		String render;
+		int color;
+		if (!bedwarsPlayer.isAlive()) {
+			if (bedwarsPlayer.isDisconnected()) {
+				return;
+			}
+			int tickTillLive = Math.max(0, bedwarsPlayer.getTickAlive() - this.client.inGameHud.getTicks());
+			float secondsTillLive = tickTillLive / 20f;
+			render = String.format("%.1f", secondsTillLive) + "s";
+			color = new Color(200, 200, 200).getAsInt();
+		} else {
+			int health = objective.getScoreboard().getPlayerScore(player, objective).getScore();
+			color = Color.blend(new Color(255, 255, 255), new Color(215, 0, 64), (int) (1 - (health / 20f) * 100)).getAsInt();
+			render = String.valueOf(health);
+		}
+		// Health
+		this.client.textRenderer.drawWithShadow(matrices,
+			render,
+			(float) (endX - this.client.textRenderer.getWidth(render)),
+			(float) y,
+			color
+		);
+
+	}
+
+	@ModifyVariable(
+		method = "render",
+		at = @At(
+			value = "STORE"
+		),
+		ordinal = 7
+	)
+	public int axolotlclient$changeWidth(int value) {
+		if (BedwarsMod.getInstance().isEnabled() && BedwarsMod.getInstance().blockLatencyIcon() && (BedwarsMod.getInstance().isWaiting() || BedwarsMod.getInstance().inGame())) {
+			value -= 9;
+		}
+		if (BedwarsMod.getInstance().isEnabled() && BedwarsMod.getInstance().isWaiting()) {
+			value += 20;
```

Comment:
This is a magic number for making sure the stars display well enough. This could maybe be changed into an actual accurate width value.

Reply:
that's not really possible without jank.. (since we don't have access to any string here)